pennylane/estimator/templates/subroutines.py

```diff
@@ -971,6 +971,204 @@ def resource_decomp(cls, base_cmpr_op, num_iter):
+        return CompressedResourceOp(cls, num_wires, params)
+
+    @classmethod
+    def resource_decomp(
+        cls,
+        unitary: CompressedResourceOp,
+        num_iterations: int,
+        adj_qft_cmpr_op: CompressedResourceOp | None = None,
+    ) -> list[GateCount | Allocate | Deallocate]:
+        r"""Returns the resources for Quantum Phase Estimation implemented using unary iteration.
+
+        This method calculates the resources required for QPE where the controlled-unitary operations
+        are replaced with a Select operation implemented using unary iteration.
+
+        The decomposition of the unary iteration follows the construction described in Fig. 2 of Section III in
+        `Babbush et al. (2011.03494) <https://arxiv.org/abs/2011.03494>`_.
+
+        Args:
+            unitary (~.pennylane.estimator.resource_operator.CompressedResourceOp): A compressed resource operator
+                corresponding to the phase estimation operator :math:`U`.
+            num_iterations (int): The total number of times the phase estimation operator :math:`U` is
+                applied in order to reach a target precision in the eigenvalue estimate.
+            adj_qft_cmpr_op (~.pennylane.estimator.resource_operator.CompressedResourceOp | None): An optional compressed
+                resource operator corresponding to the adjoint QFT routine. If ``None``, the
+                default :class:`~.pennylane.estimator.templates.subroutines.QFT` will be used.
+
+        Returns:
+            list[:class:`~.pennylane.estimator.resource_operator.GateCount`]: A list of ``GateCount`` objects, where each object
+            represents a specific quantum gate and the number of times it appears
+            in the decomposition.
+        """
+        num_wires = int(math.ceil(math.log2(num_iterations + 1)))
+
+        hadamard = resource_rep(qre.Hadamard)
+        x = resource_rep(qre.X)
+        cnot = resource_rep(qre.CNOT)
+        ctrl_unitary = resource_rep(
+            qre.Controlled, {"base_cmpr_op": unitary, "num_ctrl_wires": 1, "num_zero_ctrl": 0}
+        )
+        left_elbow = resource_rep(qre.Toffoli, {"elbow": "left"})
+        right_elbow = resource_rep(qre.Toffoli, {"elbow": "right"})
+
+        if adj_qft_cmpr_op is None:
+            adj_qft_cmpr_op = resource_rep(
+                qre.Adjoint,
+                {
+                    "base_cmpr_op": resource_rep(QFT, {"num_wires": num_wires}),
+                },
+            )
+
+        return [
+            Allocate(num_wires - 1),
+            GateCount(hadamard, num_wires),
+            GateCount(left_elbow, num_iterations - 1),
+            GateCount(cnot, num_iterations - 1),
+            GateCount(x, 2 * (num_iterations - 1)),
+            GateCount(ctrl_unitary, num_iterations),
+            GateCount(right_elbow, num_iterations - 1),
+            GateCount(adj_qft_cmpr_op),
+            Deallocate(num_wires - 1),
```

Comment:
I think these are the correct numbers based on the reference you shared. Also I thought we agreed that we needed powers of ctrl_unitaries in general? 

```suggestion
            Allocate(num_wires - 2),
            GateCount(hadamard, num_wires),
            GateCount(left_elbow, num_iterations - 2),
            GateCount(cnot, num_iterations - 2),
            GateCount(x, 2 * (num_iterations - 1)),
            GateCount(ctrl_unitary, num_iterations),
            GateCount(right_elbow, num_iterations - 2),
            GateCount(adj_qft_cmpr_op),
            Deallocate(num_wires - 1),
```

Reply:
I think the original numbers are correct. I double checked with the image: when W=10, we can see there's t0, t1, t2, t3 (num_wires = 4) and 3 ancillas. Then we can just count all left elbows/cnots, etc., to find that they're all 9.